sl_functionality.cc

```diff
@@ -200,6 +263,41 @@ void ServiceLayerFunctionality::broadcast_chirp(const std::string& username, Chi
       client_->put(follower_key, monitor_serial);
     }
   }
+
+  std::string message = chirp.text();
```

Comment:
Perhaps break out this substring parsing to a separate, individually testable string parsing function?

Reply:
I forgot to fork the repository instead of just being on Krishna's repo but done on the forked version (new pr to come I am so sorry)